### PR TITLE
- NN bump to v32.0.0; - `E-Horizon` `Edge`: added country code `ISO 3166-1 alpha-3`

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
- NN bump to v32.0.0; 
- `E-Horizon` `Edge`: added country code `ISO 3166-1 alpha-2`.

Cherry picked https://github.com/mapbox/mapbox-navigation-android/pull/4154

### Changelog
```
<changelog>Added `Edge` `countryCodeIso2`</changelog>
<changelog>Renamed `Edge` `countryCode`->`countryCodeIso3`</changelog>
<changelog>Fixed `waypoint_names` are skipped in offline routing</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt [hunk 1]
```diff
@@ -52,7 +53,8 @@ class Edge internal constructor(
     val speedLimit: Double?,
     val laneCount: Byte?,
     val meanElevation: Double?,
-    val countryCode: String?,
+    val countryCodeIso2: String?,
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt:56]: cc @mskurydin @SiarheiFedartsou for visibility

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizonMapper.kt [hunk 2]
```diff
@@ -70,6 +70,7 @@ internal object EHorizonMapper {
             speedLimit,
             laneCount,
             meanElevation,
+            countryCodeIso2,
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizonMapper.kt:73]: @mskurydin @SiarheiFedartsou could you also rename `countryCode` -> `countryCodeIso3` on NN side

File: gradle/dependencies.gradle [hunk 3]
```diff
@@ -11,7 +11,7 @@ ext {
       mapboxSdkServices         : '5.9.0-alpha.1',
       mapboxEvents              : '6.2.2',
       mapboxCore                : '3.1.1',
-      mapboxNavigator           : '31.0.1',
+      mapboxNavigator           : '32.0.0',
       mapboxCommonNative        : '9.1.0',
```
Comment (LukasPaczos) [inline gradle/dependencies.gradle:15]: This should use version 9.2.0.

Comment (RingerJK) [reply gradle/dependencies.gradle:15]: nice catch

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt [hunk 4]
```diff
@@ -52,7 +53,8 @@ class Edge internal constructor(
...
+    val countryCodeIso3: String?,
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt:57]: For consistency, could we change the order here? In 2.0 we have `countryCodeIso3` first 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/f4e3e9c2fd7cf4cda7401609807af3ff0dc548b2/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizonEdgeMetadata.kt#L43-L44

File: libnavigation-core/api/current.txt [hunk 5]
```diff
@@ -539,7 +539,8 @@ package com.mapbox.navigation.core.trip.model.eh {
 
   public final class Edge {
     method public boolean getBridge();
-    method public String? getCountryCode();
+    method public String? getCountryCodeIso2();
```
Comment (LukasPaczos) [inline libnavigation-core/api/current.txt:542]: This is a major API-breaking change.

We should leave `getCountryCode`, but wire it to `getCountryCodeIso2` and deprecate.

Comment (LukasPaczos) [reply libnavigation-core/api/current.txt:542]: or wire to `getCountryCodeIso3`, not sure which one was returned earlier

**Review by Guardiola31337 — COMMENTED** [review]
Could we also add the `@Deprecated` annotation to EH APIs as part of this PR?

cc @etl @truburt @korshaknn

File: libnavigation-core/api/current.txt (hunk 5 above)
Comment (Guardiola31337) [reply libnavigation-core/api/current.txt:542]: I'd say it was `countryCodeIso3` https://github.com/mapbox/mapbox-navigation-android/pull/4182#discussion_r602231457

Could you confirm @mskurydin @SiarheiFedartsou @korshaknn?

Comment (RingerJK) [reply libnavigation-core/api/current.txt:542]: `getCountryCodeIso3` is actually `countryCode`, and new one is `countryCodeIso2`

cc @mskurydin

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt [hunk 6]
```diff
@@ -152,7 +163,8 @@ class Edge internal constructor(
             "speedLimit=$speedLimit, " +
             "laneCount=$laneCount, " +
             "meanElevation=$meanElevation, " +
-            "countryCode=$countryCode, " +
+            "countryCodeIso2=$countryCodeIso2, " +
+            "countryCodeIso3=$countryCodeIso3, " +
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt:167]: I'd change the order here too to match the class order.

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**